cudamapper/src/overlapper.cpp

```diff
@@ -17,6 +17,45 @@
 #include <claragenomics/utils/signed_integer_utils.hpp>
 #include <mutex>
 #include <future>
+namespace
+{
+bool overlaps_mergable(const claragenomics::cudamapper::Overlap o1, const claragenomics::cudamapper::Overlap o2)
+{
+    bool relative_strands_forward = (o2.relative_strand == claragenomics::cudamapper::RelativeStrand::Forward) && (o1.relative_strand == claragenomics::cudamapper::RelativeStrand::Forward);
+    bool relative_strands_reverse = (o2.relative_strand == claragenomics::cudamapper::RelativeStrand::Reverse) && (o1.relative_strand == claragenomics::cudamapper::RelativeStrand::Reverse);
+
+    if (!(relative_strands_forward || relative_strands_reverse))
+    {
+        return false;
+    }
+
+    bool ids_match = (o1.query_read_id_ == o2.query_read_id_) && (o1.target_read_id_ == o2.target_read_id_);
+
+    if (!ids_match)
+    {
+        return false;
+    }
+
+    int query_gap = (o2.query_start_position_in_read_ - o1.query_end_position_in_read_);
```

Comment:
see above. I thought about this more and no, even in the case of `relative_strands_reverse == true` this is not a problem.

Reply:
Then I would write down these assumptions of this function as asserts.
If any of these assumptions is violated for some reason (e.g. we change the way overlapper works at some point in the future), it is very helpful to be reminded that this function will not work properly.

Reply:
Yes, I also don't see a problem with `assert`. It has no cost in release builds, but once you see some problems with results you can run debug build and it will hopefully point you directly to error/violated assumption